pkg/controller/datavolume/import-controller.go

```diff
@@ -160,23 +184,50 @@ func (r *ImportReconciler) sync(log logr.Logger, req reconcile.Request) (dvSyncR
 }
 
 func (r *ImportReconciler) syncImport(log logr.Logger, req reconcile.Request) (dvSyncState, error) {
-	syncState, syncErr := r.syncCommon(log, req, nil, nil)
+	syncState, syncErr := r.syncCommon(log, req, r.cleanup, nil)
 	if syncErr != nil || syncState.result != nil {
 		return syncState, syncErr
 	}
-	if err := r.handlePvcCreation(log, &syncState, r.updateAnnotations); err != nil {
+
+	pvcModifier := r.updateAnnotations
+	if syncState.usePopulator {
+		err := r.createVolumeImportSourceCR(&syncState)
+		if err != nil {
+			return syncState, err
+		}
+		pvcModifier = r.updatePVCForPopulation
+	}
+
+	if err := r.handlePvcCreation(log, &syncState, pvcModifier); err != nil {
 		syncErr = err
 	}
+
 	if syncState.pvc != nil && syncErr == nil {
 		r.setVddkAnnotations(&syncState)
 		syncErr = r.maybeSetPvcMultiStageAnnotation(syncState.pvc, syncState.dvMutated)
 	}
 	return syncState, syncErr
 }
 
+func (r *ImportReconciler) cleanup(syncState *dvSyncState) error {
+	dv := syncState.dvMutated
+	// This cleanup should be done if dv is marked for deletion, dv is succeeded
+	// and data volume is using the import populator
+	if dv.DeletionTimestamp == nil && dv.Status.Phase != cdiv1.Succeeded && !syncState.usePopulator {
```

Comment:
this seems complicated to me.  If the CR is owned by the dv, then only explicitly delete when the target PVC is bound?  Seems simpler to me

Reply:
it is redundant to do the delete if we don't use populators, and phase succeded is basically the same as target PVC is bound, so you say to remove the deletionTimestamp check when I add the ownerRef, then sure I guess we can